lib/adbc_connection.ex

```diff
@@ -365,20 +379,37 @@ defmodule Adbc.Connection do
 
   defp maybe_dequeue(state), do: state
 
-  defp handle_command({:query, query, params}, conn) do
-    with {:ok, stmt} <- Adbc.Nif.adbc_statement_new(conn),
-         :ok <- Adbc.Nif.adbc_statement_set_sql_query(stmt, query),
+  defp handle_command({:query, query, params}, conn) when is_binary(query) do
+    with stmt <- create_statement(conn, query),
          :ok <- maybe_bind(stmt, params) do
       Adbc.Nif.adbc_statement_execute_query(stmt)
     end
   end
 
+  defp handle_command({:query, stmt, params}, _conn) when is_reference(stmt) do
+    with :ok <- maybe_bind(stmt, params) do
+      Adbc.Nif.adbc_statement_execute_query(stmt)
+    end
+  end
+
   defp handle_command({name, args}, conn) do
     with {:ok, stream_ref} <- apply(Adbc.Nif, name, [conn | args]) do
       {:ok, stream_ref, -1}
     end
   end
 
+  defp create_statement(conn, query) when is_pid(conn) do
+    %{conn: conn} = :sys.get_state(conn)
```

Comment:
`:sys.get_state` is used for testing and debugging. In actual code, we should rather send a message to the process. But no worries, I will refactor it after merging. :)